crates/bevy_asset/src/assets.rs

```diff
@@ -549,23 +549,23 @@ impl<A: Asset> Assets<A> {
         while let Ok(drop_event) = assets.handle_provider.drop_receiver.try_recv() {
             let id = drop_event.id.typed();
 
-            assets.queued_events.push(AssetEvent::Unused { id });
-
-            let mut remove_asset = true;
-
             if drop_event.asset_server_managed {
-                let untyped_id = drop_event.id.untyped(TypeId::of::<A>());
+                let untyped_id = id.untyped();
                 if let Some(info) = infos.get(untyped_id) {
                     if let LoadState::Loading | LoadState::NotLoaded = info.load_state {
                         not_ready.push(drop_event);
                         continue;
                     }
                 }
-                remove_asset = infos.process_handle_drop(untyped_id);
-            }
-            if remove_asset {
-                assets.remove_dropped(id);
+
+                if !infos.process_handle_drop(untyped_id) {
+                    // a new handle has been created, or the asset doesn't exist
```

Comment:
Can we expand this comment a bit with the context you put in your last comment to me? That this is a check to see if a new handle has been created in the same frame after the last one was dropped, meaning it's not _actually_ unused.

Also, what does "or the asset doesn't exist" mean here?

Reply:
it's what the comment in the process_handle_drop function says. i don't know a scenario where it would happen.